src/relayclient/DevRelayClient.ts

```diff
@@ -0,0 +1,254 @@
+    return super.relayTransaction(gsnTransactionDetails)
+  }
+
+  // stop background relay
+  async stopRelay (): Promise<void> {
+    if (!this.serverStarted) {
+      return
+    }
+    if (this.httpServer !== undefined) {
+      this.httpServer.stop()
+      this.httpServer = undefined
+    }
+    if (this.relayServer !== undefined) {
+      // @ts-ignore
+      await this.relayServer.txStoreManager.clearAll()
+      this.relayServer.stop()
+      this.relayServer = undefined
+    }
+  }
+
+  /**
+     * initialize a local relay
+     * @private
+     */
+  async _initializeRelay (): Promise<void> {
+    if (this.serverStarted) {
+      return
+    }
+
+    const hub = await this.contractInteractor._createRelayHub(this.config.relayHubAddress)
+    const stakeManagerAddress = await hub.getStakeManager()
+
+    // flag early, so only the first call will try to bring up a relay
+    // (TODO: other calls should still wait for the relay to start)
+    this.serverStarted = true
+
+    // const {
+    //   relayOwner, relayWorkdir, relayHub, relayListenPort,
+    //   baseRelayFee, pctRelayFee, gasPriceFactor, devMode
+    // } = this.config
+
+    const web3provider = this.contractInteractor.provider
+    const { httpServer, relayServer } = await runServer(
+      web3provider,
+      this.config.relayHubAddress,
+      this.devConfig
+    )
+    this.relayServer = relayServer
+    this.httpServer = httpServer
+
+    // @ts-ignore
+    const IStakeManagerContract: Contract<IStakeManagerInstance> = TruffleContract({
+      contractName: 'IStakeManager',
+      abi: stakeManagerAbi
+    })
+    IStakeManagerContract.setProvider(this.contractInteractor.provider, undefined)
+
+    const stakeManager = await IStakeManagerContract.at(stakeManagerAddress)
+
+    const estim = await stakeManager.contract.methods.stakeForAddress(relayServer.address, weekInSec)
+      .estimateGas({ value: 1e18 })
+    this.debug('== staking relay gas estim:', estim)
+    await stakeManager.stakeForAddress(relayServer.address, weekInSec, {
+      from: this.devConfig.relayOwner,
+      value: ether('1'),
+      gas: estim
+    })
+    this.debug('== sending balance to relayServer', relayServer.address)
+    await stakeManager.authorizeHub(relayServer.address, this.config.relayHubAddress, { from: this.devConfig.relayOwner })
+    await this.contractInteractor.web3.eth.sendTransaction({
+      from: this.devConfig.relayOwner,
+      to: relayServer.address,
+      value: 1e18
+    })
+    this.debug('== waiting for relay')
+    // @ts-ignore
+    const relayInfo = await waitForRelay(relayServer.url as string + '/getaddr', 5000, (res) => {
+      if (res?.data?.Ready === true) {
+        return {
+          ...res.data,
+          pctRelayFee: '0',
+          baseRelayFee: '0',
+          // @ts-ignore
+          relayUrl: relayServer.url
+        }
+      }
+    })
+
+    const devRelays = this.knownRelaysManager as DevKnownRelays
+    devRelays.devRelay = relayInfo
```

Comment:
It is not a pay-per-word message, and I bet your screen can handle more then 480 characters in a row :-)